internal/command/apps/create.go

```diff
@@ -71,15 +72,18 @@ func RunCreate(ctx context.Context) (err error) {
 		fName         = flag.GetString(ctx, "name")
 		fGenerateName = flag.GetBool(ctx, "generate-name")
 		apiClient     = client.FromContext(ctx).API()
+		ctxName       = appconfig.NameFromContext(ctx)
 	)
 
 	var name string
 	switch {
-	case aName != "" && fName != "" && aName != fName:
-		err = fmt.Errorf("two app names specified %s and %s, only one may be specified",
-			aName, fName)
+	case aName != "" && fName != "" && areNamesClashing([]string{aName, fName, ctxName}):
```

Comment:
I'd pass the list of names to [lo.Compact](https://github.com/samber/lo#compact) and fail if its length is not equal 1.